3_visualize/src/sf_utils_shift.R

```diff
@@ -0,0 +1,287 @@
+
+# Utility functions for shifting
+
+#' @title extract and define shifting criteria
+#' @description defines and extracts shifting criteria
+#' for six different regions. If the input to `region_abbr`
+#' does not match existing definitions, then shifting values
+#' are returned which will do nothing to the spatial feature.
+#' @param region_abbr a single character string for which
+#' region to pull out shifting information for.
+get_shift <- function(region_abbr){
+  switch(
+    region_abbr,
+    AK = list(scale = 0.75, shift = c(-30,-200), rotation_deg = -40),
+    HI = list(scale = 1.75, shift = c(300, 0), rotation_deg = -35),
+    PR = list(scale = 3.15, shift = c(-95,75), rotation_deg=20),
+    VI = list(scale = 4, shift = c(-115,110), rotation_deg=20),
+    GU = list(scale = 5, shift = c(725, -575), rotation_deg=-90),
+    AS = list(scale = 5, shift = c(675, -30), rotation_deg=-45),
+    MP = list(scale = 1.5, shift = c(815,-600), rotation_deg=0),
+    # If there is no match, default to doing nothing to the object
+    list(scale = 1, shift = c(0,0), rotation_deg=0)
+  )
+}
+
```

Comment:
long-term it would be cool to have a pre-defined layout options for common use basemaps. For example, if there is data to show on any of the islands, we might want them to be bigger and more visible, whereas when no data are present we might not need them to take up as much space and could scale accordingly